src/vs/editor/contrib/hover/browser/hover.ts

```diff
@@ -361,6 +362,7 @@ class ShowOrFocusHoverAction extends EditorAction {
 	}
 
 	public run(accessor: ServicesAccessor, editor: ICodeEditor, args: any): void {
+		const commandService = accessor.get(ICommandService);
```

Comment:
Maybe we could put this line after the `editor.hasModel()` check which does an early return if there is no editor model. It could be put closer to line where the actual service is used?

Reply:
oh actually, we cannot reference the ServicesAccessor after the first line

Reply:
Generally it is recommended to access the services as early as possible. In particular the accessor should not leak out of the run method (e.g. through an await or by capturing it), as it becomes invalid synchronously after the run method finishes.